page_analyzer/utils/url.py

```diff
@@ -18,3 +19,11 @@ def normalize(url_name: str) -> str:
         port = ":" + str(p.port)
         normalized = normalized.replace(port, "")
     return normalized
+
+
+def get_url_status_code(url_name: str) -> int | None:
+    try:
+        r = requests.get(url_name)
+    except requests.exceptions.ConnectionError:
```

Comment:
I'd recommend catching `RequestException` as it's lower level and will work for more error types (e.g. if timeout happens)